### DISCUSSION TITLE
Problema con ejercicio numero 1 del laboratorio 02.02 (Matriz de Cauchy)

### DISCUSSION THREAD

**Discussion by CaarlosBolanos** [rramosp/ai4eng.v1#222] [UDEA Students]
Buenas noches, necesito ayuda con el ejercicio de la matriz de cauchy ya que yo hice un programa para su respectiva solución pero al momento de subirlo y evaluarlo en la plataforma me salen unos errores que no comprendo como solucionar, el error es este:

error: ERROR EXECUTING YOUR CODE
---------------------------------
```
Traceback (most recent call last):
  File "calling_script.py", line 24, in <module>
    r = eval("ggkfbpmi.{}(source_functions, source_variables, '[calberto.bolanos@udea.edu.co](mailto:calberto.bolanos@udea.edu.co)')".format(grader_function_name), namespace)
  File "<string>", line 1, in <module>
  File "/content/ggkfbpmi.py", line 22, in grader_01
    answer = st_cauchy(x,y)
  File "<string>", line 7, in cauchy
NameError: name 'x' is not defined
 (set session.debug=True for tracebacks)
```
Intenté de muchas maneras solucionarlo, sin embargo no he podido, no se si estoy entendiendo mal el problema y por eso me sale dicho error y pues finalmente este es mi código:
```
import numpy as np
x = np.array([35, 21, 57,65, 44])
y = np.array([27, 17, 25, 25, 28])

def cauchy(a,b):
    a=np.zeros((5,5))
    for i in range(0,5):
        for j in range(0,5):
            try:
                np.seterr(divide='raise')
                a[i,j]=1/(x[i]-y[j])
            except FloatingPointError:
                return("Division entre cero")
    return(a)
print(cauchy(x,y))
```
Siendo (x) y (y) los dos vectores.
Quedo atento a su respuesta, gracias.

**Comment by rramosp** [comment]
estas usando la variable 'x' que está definida fuera de la función del
ejercicio. .... el corrector solo ve lo que hay dentro de la función ... tu
función sólo debería de usar las variables a y b que son los argumentos

On Tue, Feb 14, 2023 at 9:00 PM CaarlosBolanos ***@***.***>
wrote:

> Buenas noches, necesito ayuda con el ejercicio de la matriz de cauchy ya
> que yo hice un programa para su respectiva solución pero al momento de
> subirlo y evaluarlo en la plataforma me salen unos errores que no comprendo
> como solucionar, el error es este:
> error: ERROR EXECUTING YOUR CODE
>
> Traceback (most recent call last):
>
>   File "calling_script.py", line 24, in <module>
>
>     r = eval("ggkfbpmi.{}(source_functions, source_variables, ***@***.******@***.***)')".format(grader_function_name), namespace)
>
>   File "<string>", line 1, in <module>
>
>   File "/content/ggkfbpmi.py", line 22, in grader_01
>
>     answer = st_cauchy(x,y)
>
>   File "<string>", line 7, in cauchy
>
> NameError: name 'x' is not defined
>
>  (set session.debug=True for tracebacks)
>
>
> Intenté de muchas maneras solucionarlo, sin embargo no he podido, no se si
> estoy entendiendo mal el problema y por eso me sale dicho error y pues
> finalmente este es mi código:
>
> import numpy as np
>
> x = np.array([35, 21, 57,65, 44])
>
> y = np.array([27, 17, 25, 25, 28])
>
>
>
> def cauchy(a,b):
>
>     a=np.zeros((5,5))
>
>     for i in range(0,5):
>
>         for j in range(0,5):
>
>             try:
>
>                 np.seterr(divide='raise')
>
>                 a[i,j]=1/(x[i]-y[j])
>
>             except FloatingPointError:
>
>                 return("Division entre cero")
>
>     return(a)
>
> print(cauchy(x,y))
>
>
> Siendo (x) y (y) los dos vectores.
> Quedo atento a su respuesta, gracias.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/222>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIIFQA75GYDCOCYZYFLWXQ2FHANCNFSM6AAAAAAU4IRYZ4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Reply by CaarlosBolanos** [reply]
Profesor Raúl, buenas noches.
Ya he solucionado los diferentes errores que me aparecen al ejecutar mi código en la plataforma, sin embargo mi calificación no aparece en 5, creo que mi programa ya funciona en su totalidad y cumple con lo que se pide pero no se que me falta o porque no me da la plataforma el visto bueno, este es mi código:
```
import numpy as np
def cauchy(x,y):
    c=len(x)
    a=np.zeros((c,c))
    for i in range(0,c):
        for j in range(0,c):
            try:
                a[i,j]=1/(x[i]-y[j])
                np.seterr(divide='raise')
            except ZeroDivisionError:
                return "Division por cero"
            except FloatingPointError:
                return "Division por cero"
    return a
print(cauchy(np.array([45, 31, 67, 75, 54]),np.array([17,  7, 15, 15, 18])))
```

Quedo atento a su respuesta.

**Comment by rramosp** [comment]
qué te arroja el corrector cuando haces el submit?

On Wed, Feb 15, 2023 at 1:19 AM CaarlosBolanos ***@***.***>
wrote:

> Profesor Raúl, buenas noches.
> Ya he solucionado los diferentes errores que me aparecen al ejecutar mi
> código en la plataforma, sin embargo mi calificación no aparece en 5, creo
> que mi programa ya funciona en su totalidad y cumple con lo que se pide
> pero no se que me falta o porque no me da la plataforma el visto bueno,
> este es mi código:
>
> import numpy as np
>
> def cauchy(x,y):
>
>     c=len(x)
>
>     a=np.zeros((c,c))
>
>     for i in range(0,c):
>
>         for j in range(0,c):
>
>             try:
>
>                 a[i,j]=1/(x[i]-y[j])
>
>                 np.seterr(divide='raise')
>
>             except ZeroDivisionError:
>
>                 return "Division por cero"
>
>             except FloatingPointError:
>
>                 return "Division por cero"
>
>     return a
>
> print(cauchy(np.array([45, 31, 67, 75, 54]),np.array([17,  7, 15, 15, 18])))
>
>
> Quedo atento a su respuesta.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/222#discussioncomment-4979539>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIM5H35V3L3VKT3UB3TWXRYN5ANCNFSM6AAAAAAU4IRYZ4>
> .
> You are receiving this because you commented.Message ID: <rramosp/ai4eng.
> ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.

**Reply by CaarlosBolanos** [reply]
Profesor cuando ejecuto mi código en la plataforma el sistema me arroja esto:
```
task_01 submitted. your grade is 0
----- grader message -------
incorrect: testing with 100 random calls
your code did not correctly detect a division by zero
x = [47 68 98 28 66], y=[47 1 2 14 8]
```

Sin embargo con esos valores de (x) y (y) que me da la corrección mi código si funciona y me arroja un mensaje diciendo que hay una división por cero, entonces no entiendo cual es el problema.
```
def cauchy(x,y):

    c=len(x)
    a=np.zeros((c,c))
    for i in range(0,c):
        for j in range(0,c):
            try:
                a[i,j]=1/(x[i]-y[j])
                np.seterr(divide='raise')
            except ZeroDivisionError:
                return "Division por cero"
            except FloatingPointError:
                return "Division por cero"
    return a
print(cauchy(np.array([47 ,68 ,98 ,28 ,66]),np.array([47, 1 ,2 ,14 ,8])))
```
```
Division por cero
```

Y cada vez aparece un nuevo conjunto de números de (x) y (y) los cuales la plataforma me dicen que no cumplen pero yo los monto en mi programa y si me funcionan.

**Comment by rramosp** [comment]
fíjatte bien en el enunciado ... cuando detectas una división por cero
tienes que levantar (raise) una excepción ValueError ... pero tu código
devuelve un string, no levanta ninguna excepción

On Wed, Feb 15, 2023 at 9:28 AM CaarlosBolanos ***@***.***>
wrote:

> Profesor cuando ejecuto mi código en la plataforma el sistema me arroja
> esto:
>
> task_01 submitted. your grade is 0
>
> ----- grader message -------
>
> incorrect: testing with 100 random calls
>
> your code did not correctly detect a division by zero
>
> x = [47 68 98 28 66], y=[47 1 2 14 8]
>
>
> Sin embargo con esos valores de (x) y (y) que me da la corrección mi
> código si funciona y me arroja un mensaje diciendo que hay una división por
> cero, entonces no entiendo cual es el problema.
>
> def cauchy(x,y):
>
>
>
>     c=len(x)
>
>     a=np.zeros((c,c))
>
>     for i in range(0,c):
>
>         for j in range(0,c):
>
>             try:
>
>                 a[i,j]=1/(x[i]-y[j])
>
>                 np.seterr(divide='raise')
>
>             except ZeroDivisionError:
>
>                 return "Division por cero"
>
>             except FloatingPointError:
>
>                 return "Division por cero"
>
>     return a
>
> print(cauchy(np.array([47 ,68 ,98 ,28 ,66]),np.array([47, 1 ,2 ,14 ,8])))
>
>
> Division por cero
>
>
> Y cada vez aparece un nuevo conjunto de números de (x) y (y) los cuales la
> plataforma me dicen que no cumplen pero yo los monto en mi programa y si me
> funcionan.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/222#discussioncomment-4984032>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZILIDOMRHCIRNGCEXVDWXTRXXANCNFSM6AAAAAAU4IRYZ4>
> .
> You are receiving this because you commented.Message ID: <rramosp/ai4eng.
> ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
